### PR TITLE
`Development`: Add server tests for job module

### PR DESCRIPTION
**Pull request by ani-ib**
#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

This PR adds integration tests for the JobResource REST controller and related service logic.
The motivation is to ensure correctness, prevent regressions, and cover edge cases around job creation, updates, deletions, state changes, and retrieval.
Additionally, the deleteJob method in JobService was simplified by removing unnecessary try/catch logic.

### Description

Added a comprehensive JobResourceTest covering:
- getAvailableJobs (invalid pagination)
- updateJob (existing and non-existing job)
- deleteJob (existing and non-existing job)
- changeJobState (success + not found)
- getJobsByProfessor (valid + invalid pagination)
- getJobById (success + not found)
- getJobDetails (success + not found)
- Adjusted MvcTestClient to support:
   - Handling Void return types cleanly (deleteAndReadOk, postAndReadOk, etc.)
   - Correct expectation of 204 No Content for DELETE.
   - Updated JobService.deleteJob to remove redundant exception handling.

### Steps for Testing

Steps:
1. checkout this branch
2. run JobResourceTest
3. confirm all tests pass successfully

#### Code Review

- [ ] Code Review 1

#### Manual Tests

- [x] Test 1

### CODE REVIEW

**Review by Abi107717 — APPROVED** [review]
tested it locally, all tests passed

File: src/test/java/de/tum/cit/aet/job/web/rest/JobResourceTest.java [hunk 1]
```diff
@@ -172,13 +201,200 @@ void createJobInvalid_DoesNotPersist() {
                 entry("tasks", "tasks"),
                 entry("requirements", "req"),
                 entry("state", "PUBLISHED")
-        );
+            );
 
-        assertThatThrownBy(() ->
+            assertThatThrownBy(() ->
                 api.postAndReadOk("/api/jobs/create", invalid, JobFormDTO.class)
```
Comment (Bofan-Zhu) [inline src/test/java/de/tum/cit/aet/job/web/rest/JobResourceTest.java:207]: Some API paths don't have a trailing slash. Could you adapt that?

**Review by Bofan-Zhu — COMMENTED** [review]
Left just one small comment

File: src/test/java/de/tum/cit/aet/job/web/rest/JobResourceTest.java (hunk 1 above)
Comment (ani-ib) [reply src/test/java/de/tum/cit/aet/job/web/rest/JobResourceTest.java:207]: thanks for pointing that out, I fixed the paths with the missing slashes 😊

**Review by Bofan-Zhu — APPROVED** [review]
